examples/keras_recipes/creating_tfrecords.py

```diff
@@ -0,0 +1,193 @@
+
+
+def _int64_feature(value):
+    """Returns an int64_list from a bool / enum / int / uint."""
+    return tf.train.Feature(int64_list=tf.train.Int64List(value=[value]))
+
+
+def _float_feature_list(value):
+    """Returns a list of float_list from a float / double."""
+    return tf.train.Feature(float_list=tf.train.FloatList(value=value))
+
+
+def create_example(image, example):
+    feature = {
+        "image": _bytes_feature(image),
+        "area": _float_feature(example["area"]),
+        "bbox": _float_feature_list(example["bbox"]),
+        "category_id": _int64_feature(example["category_id"]),
+        "id": _int64_feature(example["id"]),
+        "image_id": _int64_feature(example["image_id"]),
+    }
+
+    return tf.train.Example(features=tf.train.Features(feature=feature))
+
+
+def parse_tfrecord_fn(example):
+    feature_description = {
+        "image": tf.io.FixedLenFeature([], tf.string),
+        "area": tf.io.FixedLenFeature([], tf.float32),
+        "bbox": tf.io.VarLenFeature(tf.float32),
+        "category_id": tf.io.FixedLenFeature([], tf.int64),
+        "id": tf.io.FixedLenFeature([], tf.int64),
+        "image_id": tf.io.FixedLenFeature([], tf.int64),
+    }
+    example = tf.io.parse_single_example(example, feature_description)
+    example["image"] = tf.image.decode_jpeg(example["image"], channels=3)
+    example["bbox"] = tf.sparse.to_dense(example["bbox"])
+
+    return example
+
+
+"""
+## Generating TFRecords
+
+The generated TFRecrods will have the file name format `file_{number}.tfrec`, this is
+optional, but writing the number of files at the file name can make counting easier.
+"""
+
+for tfrec_num in range(n_tfrecods):
+    samples = annotations[(tfrec_num * n_samples) : ((tfrec_num + 1) * n_samples)]
+
+    with tf.io.TFRecordWriter(
+        tfrecords_dir + "/file_%.2i-%i.tfrec" % (tfrec_num, len(samples))
+    ) as writer:
+        for sample in samples:
+            image_path = f"{images_dir}/{sample['image_id']:012d}.jpg"
+            image = tf.io.read_file(image_path).numpy()
+
+            example = create_example(image, sample)
+            writer.write(example.SerializeToString())
+
+"""
+## Looking at a generated TFRecord of a single data sample
+"""
+
+raw_dataset = tf.data.TFRecordDataset(f"{tfrecords_dir}/file_00-{n_samples}.tfrec")
+parsed_dataset = raw_dataset.map(parse_tfrecord_fn)
+
+for features in parsed_dataset.take(1):
+    for key in features.keys():
+        if key != "image":
+            print(f"{key}: {features[key]}")
+
+    print(f"Image shape: {features['image'].shape}")
+    plt.figure(figsize=(7, 7))
+    plt.imshow(features["image"].numpy())
+    plt.show()
+
+"""
+## Conclusion
+
+Now instead of reading images and annotations from different folders, we can have all the
+information of the samples reading from the same source, the TFRecords that we just
+created. This process makes storing and reading the data much more easy and efficient.
```

Comment:
We may also want to talk about storing the data remotely (like on Google Cloud) and loading/buffering the data for training from the remote location without batch-downloading it. This is a relevant use case.

Reply:
I have added this use case to the introduction, so it can complement the TPU example.